src/main/java/io/vertx/core/json/JsonArray.java

```diff
@@ -65,16 +66,27 @@ public JsonArray() {
     list = new ArrayList<>();
   }
 
+  /**
+   * Create an empty instance with initialCapacity
+   */
+  public JsonArray(int initialCapacity) {
+    list = new ArrayList<>(initialCapacity);
+  }
+
   /**
    * Create an instance from a List. The List is not copied.
    *
    * @param list the underlying backing list
    */
-  public JsonArray(List list) {
-    if (list == null) {
+  @SuppressWarnings({"rawtypes", "unchecked"})
```

Comment:
that is an avoidable breaking change, we need to keep the List constructor

Reply:
Does it really break?

I just thought widening is not a problem 🤷‍♀️